components/TabList/tab-list.tsx

```diff
@@ -74,9 +75,15 @@ export const SubTabsList = ({ tabList, pageId, selectedTab, label, textSize, onS
             role="tab"
             aria-selected={isSelected ? "true" : "false"}
             data-state={isSelected ? "active" : "inactive"}
+            tabIndex={0} // Added tabIndex for accessibility
```

Comment:
Tab index can be removed here as well. As well, the `role="tablist"` can be removed.

```suggestion
```

For the sub tab list, it's a little more complicated, though. The SubTabList supports links, but also just text which is why there is the `onKeyDown`.

When SubTab list items are links they should be housed in a `<nav />` like it currently is, but whhen it's just text, i.e. when there is an `onSelect`, we need to adjust here.

Instead of just text, it should be a button and the `onClick` and `aria-selected` should be on the button, not the div. The `onKeyDown` can be discarded as a native button element handles the <kbd>ENTER</kbd>

```diff
-            {onSelect ? tab.name : <Link href={`${pageId ? `${pageId}/` : ""}${tab.path}`}>{tab.name}</Link>}
+            {onSelect ? <button onClick_{...} aria-slected={...}>{tab.name}</button> : <Link href={`${pageId ? `${pageId}/` : ""}${tab.path}`}>{tab.name}</Link>}
```

It'll probably be easier with the current logic of showing buttons only if `onSelect`, to do something like this instead.

```typescript
export const SubTabsList = ({ tabList, pageId, selectedTab, label, textSize, onSelect }: SubTabsListProps) => {
  return onSelect ? <div role="tablist">
    {/* logic to map through buttons */}
  </div> : (
    <nav
      aria-label={label}
      className={clsx(
        "flex items-center w-max overflow-x-auto overflow-y-hidden gap-2 bg-light-slate-3 p-1 rounded",
        textSize === "small" && "text-sm"
      )}
    >
      {tabList.map((tab, index) => {
        const isSelected = selectedTab === tab.name.toLowerCase();

        return (
          <div
            key={index}
            className={clsx(isSelected && "bg-white shadow", "rounded py-1 px-2", !isSelected && "text-light-slate-11")}
          >
            <Link href={`${pageId ? `${pageId}/` : ""}${tab.path}`}>{tab.name}</Link>
          </div>
        );
      })}
    </nav>
  );
};
```

I realize this is more involved than the original issue, but this happens sometimes when digging into the problem a little deeper.

Let me know if you have any questions, and happy to assist.